components/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt/src/main/java/org/wso2/carbon/identity/api/resource/mgt/dao/impl/AuthorizationDetailsTypeMgtDAOImpl.java

```diff
@@ -0,0 +1,523 @@
+                                                                                 String typeId, Integer tenantId)
+            throws SQLException {
+
+        if (isRichAuthorizationRequestsDisabled()) {
+            return null;
+        }
+
+        try (PreparedStatement preparedStatement =
+                     dbConnection.prepareStatement(SQLConstants.GET_AUTHORIZATION_DETAILS_TYPE_BY_API_ID_AND_TYPE_ID)) {
+            preparedStatement.setString(1, apiId);
+            preparedStatement.setString(2, typeId);
+            preparedStatement.setObject(3, tenantId);
+
+            final ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                return this.buildAuthorizationDetailsType(resultSet);
+            }
+        }
+        return null;
+    }
+
+    private AuthorizationDetailsType buildAuthorizationDetailsType(final ResultSet resultSet)
+            throws SQLException {
+
+        final String authorizationDetailsTypeId = resultSet.getString(SQLConstants.ID_COLUMN_NAME);
+
+        if (StringUtils.isBlank(authorizationDetailsTypeId)) {
+            return null;
+        }
+
+        return new AuthorizationDetailsType.AuthorizationDetailsTypesBuilder()
+                .id(authorizationDetailsTypeId)
+                .type(resultSet.getString(SQLConstants.AUTHORIZATION_DETAILS_TYPE_COLUMN_NAME))
+                .name(resultSet.getString(SQLConstants.NAME_COLUMN_NAME))
+                .description(resultSet.getString(SQLConstants.DESCRIPTION_COLUMN_NAME))
+                .schema(parseSchema(resultSet.getString(SQLConstants.AUTHORIZATION_DETAILS_SCHEMA_COLUMN_NAME)))
+                .build();
+    }
+
+    private String buildGetAuthorizationDetailsTypesSqlStatement(final String filterQuery) {
+
+        return String.format(SQLConstants.GET_AUTHORIZATION_DETAILS_TYPE_BY_TENANT_ID_FORMAT,
+                filterQuery, SQLConstants.GET_SCOPES_BY_TENANT_ID_TAIL);
+    }
+
+    private void assertAuthorizationDetailsTypeExistence(Connection dbConnection, String type, Integer tenantId)
+            throws APIResourceMgtException {
+
+        if (isBlank(type)) {
+            throw APIResourceManagementUtil.handleClientException(APIResourceManagementConstants
+                    .ErrorMessages.ERROR_CODE_AUTHORIZATION_DETAILS_TYPE_EMPTY);
+        }
+
+        if (this.isAuthorizationDetailsTypeExists(dbConnection, type, tenantId)) {
+
+            if (LOG.isDebugEnabled()) {
+                LOG.debug(String.format("Authorization details type '%s' already exists for tenant.", type));
+            }
+
+            throw APIResourceManagementUtil.handleClientException(APIResourceManagementConstants
+                    .ErrorMessages.ERROR_CODE_AUTHORIZATION_DETAILS_TYPE_EXISTS, type);
+        }
+    }
+
+    private boolean isValidAuthorizationDetailsType(Connection dbConnection, String apiId,
```

Comment:
cant we move this validations to service layer?

Reply:
Added a more meaningful name to the method with 2d72c264576eb9e95bb40ef8fef70f8cdd5cbeb1